include/mbedtls/ssl_internal.h

```diff
@@ -945,6 +945,8 @@ static inline int mbedtls_ssl_get_psk( const mbedtls_ssl_context *ssl,
 
     else
     {
+        *psk = NULL;
+        *psk_len = 0;
```

Comment:
Why do you need that if you return in error?

Reply:
It's something we often do, as a sort of second line of defense: the caller is definitely supposed to check the return code, but in case it failed to do that, we also return data that's obviously invalid. In that case I'm not sure it's strictly necessary, as in common uses cases those argument will already have those values when entering the function anyway, but being too careful never hurts.

Reply:
There is a small memory footprint cost thus I am inclined not to do so in "internal" functions. But this is a discussion that goes beyond this PR thus I am resolving this conversation. I am happy for the code to stay at it is now.